packages/core/src/routes/interaction/utils/passcode-validation.ts

```diff
@@ -0,0 +1,41 @@
+import { PasscodeType } from '@logto/schemas';
+import type { Event } from '@logto/schemas';
+
+import { createPasscode, sendPasscode } from '#src/lib/passcode.js';
+import type { LogContext } from '#src/middleware/koa-log.js';
+import { getPasswordlessRelatedLogType } from '#src/routes/session/utils.js';
+
+import type { SendPasscodePayload } from '../types/guard.js';
+
+/**
+ * Refactor Needed:
+ * This is a work around to map the latest interaction event type to old PasscodeType
+ *  */
+const eventToPasscodeTypeMap: Record<Event, PasscodeType> = {
+  'sign-in': PasscodeType.SignIn,
+  register: PasscodeType.Register,
+  'forgot-password': PasscodeType.ForgotPassword,
+};
```

Comment:
If the user wants to add a phone number, what event type should we use in order to send the passcode to the new phone?

Reply:
Discussed offline